code/system/application/legacy/vanishingpoint/include/LaneMarkingsScan.h

```diff
@@ -0,0 +1,373 @@
+#ifndef LANE_MARKINGS_SCAN_H_
+#define LANE_MARKINGS_SCAN_H_
+
+#include <memory>
+
+#include "MatVisitInterface.h"
+
+namespace lmvp {
+
+class LaneMarkingsScanner : public MatVisitor {
+public:
+    /**
+     * @param scanDirection In which direction the scanner shall look for white pixels
+     * @param scanRegion The region in which to look for lane markings
+     * @param maxMarkingWidth After which maximum scan length per row the scanning shall be interrupted
+     * @param maxMarkingRowCount The maximum number of rows a marking shall span
+     * @param maxMarkingInclination The maximum inclination a road marking is expected to have,
+     *                              given in the amount of pixels to travel to the right/left from
+     *                              the first lane marking pixel when going to the next row
+     */
+    LaneMarkingsScanner(RelativeDirection scanDirection, const ScanRegion & scanRegion,
```

Comment:
Please have the implementation in separate implementation files (please apply this for all further occurrences).